ols/src/cache/postgres_cache.py

```
@@ -166,6 +181,81 @@ def _ddl_statements(self) -> list[str]:
 
+    def _mark_unhealthy(self) -> None:
+        """Mark health status as unhealthy (dual-feed from cache operations)."""
+        with self._health_lock:
+            self._health_status = False
+
+    def _connect_health(self) -> None:
+        """Establish a dedicated lightweight connection for health checks."""
+        if self._health_connection is not None:
+            try:
+                self._health_connection.close()
+            except Exception as close_err:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)